File: Java/3.Longest Substring.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
     Map <Character,Integer> Tracker = new HashMap<>();
     int max = 0;
     int pointer = 0;


     for(int i=0; i<s.length();i++){

        while(Tracker.containsKey(s.charAt(i))){

            Tracker.remove(s.charAt(pointer));

            pointer++;
        }
        Tracker.put(s.charAt(i), i);
        max = Math.max(max, i - pointer + 1);

     }

     return max;


    }
}

File: Python/3.LongestSubstring.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        tracker = set()
        max_element = 0
        pointer = 0

        for i in range(len(s)):
            while s[i] in tracker:
                tracker.remove(s[pointer])
                pointer +=1
            tracker.add(s[i])
            max_element = max(max_element, i - pointer+1)

        return max_element